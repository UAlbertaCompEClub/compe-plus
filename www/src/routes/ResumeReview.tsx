import React, { FC } from 'react';

export const ResumeReview: FC = () => {
    return <div>Resume Review</div>;
};
