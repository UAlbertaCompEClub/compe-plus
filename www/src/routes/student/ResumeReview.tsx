import { useAuth0 } from '@auth0/auth0-react';
import { Grid, Typography } from '@material-ui/core';
import React, { FC, useEffect } from 'react';
import { Route, Switch, useRouteMatch } from 'react-router-dom';

import { useStudentDispatch, useStudentSelector } from '../../redux/substores/student/studentHooks';
import getMyResumeReviews from '../../redux/substores/student/thunks/getMyResumeReviews';
import NoResumes from './resumeReview/NoResumes';
import UploadResume from './resumeReview/UploadResume';

const ResumeReview: FC = () => {
    const { path } = useRouteMatch();

    const { resumeReviews } = useStudentSelector((state) => state.resumeReview);

    const { getAccessTokenSilently } = useAuth0();
    const dispatch = useStudentDispatch();

    useEffect(() => {
        dispatch(getMyResumeReviews({ tokenAcquirer: getAccessTokenSilently }));
    }, []);

    useEffect(() => {
        // TODO: display upload button if there are no resume reviews
        console.debug(resumeReviews);
    }, [resumeReviews]);

    return (
        <div style={{ overflow: 'hidden' }}>
            <Grid container justify='center' alignItems='center' spacing={8} style={{ marginTop: '10px', minHeight: '75vh' }}>
                <Grid container item justify='center'>
                    <Typography variant='h1'>Resume Review</Typography>
                </Grid>
                <Switch>
                    <Route exact path={path}>
                        <NoResumes />
                    </Route>

                    <Route path={`${path}/upload`}>
                        <UploadResume />
                    </Route>
                </Switch>
            </Grid>
        </div>
    );
};

export default ResumeReview;
