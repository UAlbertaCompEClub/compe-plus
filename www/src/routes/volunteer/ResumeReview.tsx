import { useAuth0 } from '@auth0/auth0-react';
import React, { FC, useEffect } from 'react';

import getAvailableResumeReviews from '../../redux/substores/volunteer/thunks/getAvailableResumeReviews';
import { useVolunteerDispatch, useVolunteerSelector } from '../../redux/substores/volunteer/volunteerHooks';

const ResumeReview: FC = () => {
    const { availableResumes, reviewingResumes, isLoading } = useVolunteerSelector((state) => state.resumeReview);
    const { getAccessTokenSilently } = useAuth0();
    const dispatch = useVolunteerDispatch();

    useEffect(() => {
        console.log('do something');
        dispatch(getAvailableResumeReviews({ tokenAcquirer: getAccessTokenSilently }));
        console.log('dispatched');
    }, []);

    return (
        <>
            <h1>Currently Reviewing</h1>
            <h1>Available Resumes</h1>
            <p>It be loading {isLoading ? 'true' : 'false'}.</p>
            {availableResumes.map((r) => (
                <p key={r.id}>{r.id}</p>
            ))}
        </>
    );
};

export default ResumeReview;
