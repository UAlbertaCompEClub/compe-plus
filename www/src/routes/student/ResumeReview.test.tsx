import { shallow } from 'enzyme';
import React from 'react';

import ResumeReview from './ResumeReview';

describe('StudentResumeReview', () => {
    it('renders correctly', () => {
        const result = shallow(<ResumeReview />);

        expect(result).toMatchSnapshot();
    });
});
