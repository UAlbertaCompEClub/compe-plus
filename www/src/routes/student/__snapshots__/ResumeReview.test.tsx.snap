// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentResumeReview renders correctly 1`] = `
<div
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    justify="center"
    spacing={8}
    style={
      Object {
        "marginTop": "10px",
        "minHeight": "75vh",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
      justify="center"
    >
      <WithStyles(ForwardRef(Typography))
        variant="h1"
      >
        Resume Review
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        item={true}
        justify="center"
        xs={12}
      >
        <img
          src="add_resume.svg"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        container={true}
        item={true}
        justify="center"
        xs={12}
      >
        <WithStyles(ForwardRef(Typography))>
          You have no resumes submitted
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        container={true}
        item={true}
        justify="center"
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          color="primary"
          variant="contained"
        >
          Submit resume
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
